FakeServer/Startup.cs

```diff
@@ -130,6 +130,8 @@ public void Configure(IApplicationBuilder app, IHostingEnvironment env, IApplica
 
             app.UseMiddleware<HttpOptionsMiddleware>();
 
+            app.UseHeadMethodMiddleware();
```

Comment:
Normally Kestrel takes care of cleaning up the body in HEAD cases, so this needs to be only used when running the tests

```cs
if (env.EnvironmentName == "IntegrationTest")
            {
                app.UseHeadMethodMiddleware();
            }
```

Better way would be to add this when WebApplicationFactory is created, but no idea how to do that. Maybe it is not even possible 😄

Reply:
Done.

Reply:
Maybe raise a separate issue for improvement.

Reply:
Yes, this is a case for an own issue 👍